### ISSUE TITLE
remove a group

### ISSUE THREAD

**Issue by grahamehorner** [Azure/azure-signalr#232]
I'm looking at the API and can't find away that I can force a removal of a group; can you please advise if this is possible with the current version v1.0.0 of the service and/or if explicit removal of a group is planned or in progress

**Comment by vicancy** [comment]
With **the API**, do you mean the REST API https://github.com/Azure/azure-signalr/blob/dev/docs/rest-api.md#api?

Can you have some more detailed description about **removal of a group**? For what scenarios do you want **removing a group**?

**Comment by grahamehorner** [comment]
@vicancy yes, sorry the rest api should have been more explicit; example is an eventing notification where a device/user is interested is updates for an event, the user/devices subscribers to the event (which is a signalr group) and receives messages; the event is close/finished and the group removed terminating/disconnecting the clients. other example would be a chat group where the group is kicked by an administrator etc.

**Comment by davidfowl** [comment]
You can't "kick" a group. It's an ephemeral grouping of users/connections.

**Comment by grahamehorner** [comment]
@davidfowl so to remove all uses in a group, you would therefore need to send a group message to all users/connections to disconnect from the group; and have the client then remove its user/connection  from that group ? as this can’t be done server/service side?

**Comment by grahamehorner** [comment]
@davidfowl and the same would be needed to remove a user/connection from all groups ?

**Comment by davidfowl** [comment]
Yes

**Comment by grahamehorner** [comment]
@davidfowl do you think/feel that this should be functionality with-in the #SignalR clients? IMHO I think this should be baked into the client as my gut feel it that this would cover a large number of use cases, and having everyone create their own client functions to kick/remove a connection from a group or a connection from all group should be avoided where possible; also thinking that the client functionality could be extended with custom event handlers, but not overridden in terms of the disconnection.

**Comment by grahamehorner** [comment]
Yeah, thinking that the protocol should support the server termination of a client from a group, and/the termination of the user connection(s)

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by davidfowl** [comment]
There’s no concept on a group in the protocol. That’s the point. It’s a purely server side construct used to send data to a group of connections. I don’t really understand the request.

**Comment by bmmathe** [comment]
Makes sense to me.  If you have logic in your service that should terminate a group then why rely on the clients to terminate themselves?  Say for example you are building an app with a session between two or more clients.  Once the server deems the session finished, all client connections should be forcefully terminated so the server can clean up the resources.
